library/src/main/java/j2html/rendering/IndentedHtml.java

```diff
@@ -0,0 +1,247 @@
+import j2html.utils.Indenter;
+import j2html.utils.TextEscaper;
+
+import java.io.IOException;
+import java.util.Stack;
+
+/**
+ * Composes HTML with lines breaks and indentation between tags and text.
+ *
+ * @param <T> The type of the Appendable to which HTML will be appended.
+ */
+public class IndentedHtml<T extends Appendable> implements HtmlBuilder<T> {
+
+    /**
+     * Returns an HtmlBuilder that will generate indented HTML using
+     * Config defaults.
+     *
+     * @param out The Appendable to which HTML will be appended.
+     * @param <T> The type of the Appendable to which HTML will be appended.
+     * @return An HtmlBuilder for indented HTML.
+     */
+    public static final <T extends Appendable> IndentedHtml into(T out) {
+        return new IndentedHtml(out, Config.defaults());
+    }
+
+    /**
+     * Returns an HtmlBuilder that will generate indented HTML using
+     * the given Config.
+     *
+     * @param out    The Appendable to which HTML will be appended.
+     * @param config The Config which will specify indentation, text escapement, tag closing, etc.
+     * @param <T>    The type of the Appendable to which HTML will be appended.
+     * @return An HtmlBuilder for indented HTML.
+     */
+    public static final <T extends Appendable> IndentedHtml into(T out, Config config) {
+        return new IndentedHtml(out, config);
+    }
+
+    /**
+     * Returns an HtmlBuilder that will generate indented HTML in memory using
+     * Config defaults.
+     *
+     * @param <T> The type of the Appendable to which HTML will be appended.
+     * @return An HtmlBuilder for indented HTML.
+     */
+    public static final <T extends Appendable> IndentedHtml inMemory() {
+        return into(new StringBuilder());
+    }
+
+    /**
+     * Returns an HtmlBuilder that will generate indented HTML in memory using
+     * the given Config.
+     *
+     * @param config The Config which will specify indentation, text escapement, tag closing, etc.
+     * @param <T>    The type of the Appendable to which HTML will be appended.
+     * @return An HtmlBuilder for indented HTML.
+     */
+    public static final <T extends Appendable> IndentedHtml inMemory(Config config) {
+        return into(new StringBuilder(), config);
+    }
+
+    private final T out;
+    private final Indenter indenter;
+    private final TextEscaper textEscaper;
+    private final TagBuilder enclosingElementAttributes;
+    private final TagBuilder emptyElementAttributes;
+
+    // Dealing with preformatted elements (pre and textarea) requires
+    // that we know what our parent elements are.  To do that we use
+    // a stack; adding items as start tags are created, and removing them
+    // as those tags are closed.  Determining whether or not we are
+    // currently rendering into a preformatted element is as simple as
+    // asking if any tags on the stack match a preformatted element name.
+    private Stack<String> trace = new Stack<>();
```

Comment:
`Stack` is a Java 1.0 class (like `Vector`) and has `synchronized` methods which are not necessary here.
Its JavaDoc recommends using `Deque` instead so I would change this to
```suggestion
    private final Deque<String> trace = new ArrayDeque<>();
```

Reply:
Good call.  Thanks for reminding me about Stack.